providers/gcs/gcs.go

```diff
@@ -76,12 +94,38 @@ func NewBucketWithConfig(ctx context.Context, logger log.Logger, gc Config, comp
 			return nil, errors.Wrap(err, "failed to create credentials from JSON")
 		}
 		opts = append(opts, option.WithCredentials(credentials))
+	} else {
+		opts = append(opts, option.WithoutAuthentication())
```

Comment:
This change is described as "Fix tests"; it stops the GCS client from using config from environment variables.

Reply:
+1, From what I understand, this would break GCP workload identity from working

Context: https://cloud-native.slack.com/archives/CK5RSSC10/p1633961188039700

Reply:
Literally came to this repo to report this bug; this just hit me in a pre-prod environment and came to report it here :)

Reply:
@JoaoBraveCoding would you have time to look into this?

Reply:
Had plans to fix it this morning but @saswatamcode beat me to it (thank you!) Apologies for having broken GCS authentication with env vars!